--- a/drivers/amlogic/wifi/broadcom/broadcm_4354/bcmsdh_sdmmc.c~	2015-08-14 08:43:36.000000000 +0200
+++ b/drivers/amlogic/wifi/broadcom/broadcm_4354/bcmsdh_sdmmc.c	2015-08-16 11:34:45.413731514 +0200
@@ -60,6 +60,7 @@ static void IRQHandlerF2(struct sdio_fun
 #endif /* !defined(OOB_INTR_ONLY) */
 static int sdioh_sdmmc_get_cisaddr(sdioh_info_t *sd, uint32 regaddr);
 #if defined(CUSTOMER_HW_AMLOGIC)
+#define sdio_reset_comm _hidden_sdio_reset_comm
 int sdio_reset_comm(struct mmc_card *card)
 {
 	return 0;
