diff --git a/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c b/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
index 3e61dd3..bb6d3c2 100755
--- a/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
+++ b/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
@@ -108,6 +108,12 @@ const static char *bcm4339a0ag_fw_name[] = {
 	"fw_bcm4339a0_ag_p2p.bin",
 	"fw_bcm4339a0_ag_mfg.bin"
 };
+
+const static char *bcm4339a0eag_fw_name[] = {
+        "fw_bcm4339a0e_ag.bin",
+        "fw_bcm4339a0e_ag_apsta.bin",
+        "fw_bcm4339a0e_ag_p2p.bin"
+};
 
 const static char *nv_name[] = {
 	"ap6210_nvram.txt",
@@ -387,7 +393,7 @@ dhd_conf_set_nv_name_by_chip(dhd_pub_t *dhd, char *dst, char *src)
 		case BCM4335_CHIP_ID: //bcm4339a0ag BCM4335A0
 					strcpy(&dst[i+1], nv_name[3]);
 						break;
-		case BCM4339_CHIP_ID: //bcm4339a0ag
+		case BCM4339_CHIP_ID: //bcm4339a0eag
 					strcpy(&dst[i+1], nv_name[3]);
 						break;
 	}
@@ -475,7 +481,7 @@ dhd_conf_set_fw_name_by_chip(dhd_pub_t *dhd, char *dst, char *src)
 			break;
 		case BCM4339_CHIP_ID:
 			if (chiprev == BCM4339A0_CHIP_REV)
-				strcpy(&dst[i+1], bcm4339a0ag_fw_name[fw_type]);
+				strcpy(&dst[i+1], bcm4339a0eag_fw_name[fw_type]);
 			break;
 	}
 
