--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd	2015-03-31 12:11:00.000000000 +0300
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd	2015-04-23 14:45:24.447152521 +0300
@@ -314,7 +314,7 @@
 //$$ L2 PROP_STR = "status"
     rtc{
 		compatible = "amlogic,aml_rtc";
-		status = "okay";
+		status = "disabled";
 	};
 
 /// ***************************************************************************************
@@ -477,7 +477,7 @@
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            gpio_ro = "GPIOY_15";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -495,7 +495,6 @@
 //$$ L3 PROP_U32 = "max_req_size"
 //$$ L2 PROP_STR = "gpio_dat3"
 //$$ L2 PROP_U32 = "card_type"
-
 //        emmc{
 //            status = "ok";
 //            port = <2>;          /**0:sdio_a, 1:sdio_b, 2:sdio_c, 3:sdhc_a, 4:sdhc_b, 5:sdhc_c */
@@ -1895,7 +1895,7 @@
 /// ***************************************************************************************
 ///	-	Ethernet
         ethmac: ethernet@0xc9410000{
-                compatible = "amlogic,meson8m2-rmii-dwmac", "snps,dwmac";
+                compatible = "amlogic,meson8m2-rgmii-dwmac", "snps,dwmac";
                 reg = <0xc9410000 0x10000
                        0xc1108108 0x4>;
                 interrupts = <0 8 1>;
